File: scripts/update-assistant.ts
import dotenv from "dotenv";
import { MICHELLE_PROMPT_V9, RESPONSE_FORMAT_V9 } from "../src/lib/ai/prompt";
import fetch from "node-fetch";

dotenv.config({ path: ".env.local" });

const API_KEY = process.env.OPENAI_API_KEY;
// Assistant ID (v9.0 Michelle Psychology Brand)
const ASSISTANT_ID = "asst_h5rrljLWogiiDUrgzz0hH17C";

if (!API_KEY) {
  console.error("❌ OPENAI_API_KEY is missing.");
  process.exit(1);
}

const headers = {
  "Authorization": `Bearer ${API_KEY}`,
  "Content-Type": "application/json",
  "OpenAI-Beta": "assistants=v2",
};

async function updateAssistant() {
  console.log(`🔄 Updating Assistant: ${ASSISTANT_ID}...`);
  
  const instructions = `${MICHELLE_PROMPT_V9}\n\n${RESPONSE_FORMAT_V9}`;
  
  const payload = {
    instructions: instructions,
    name: "Michelle AI (v9.0 - Michelle Psychology Brand)",
  };

  const res = await fetch(`https://api.openai.com/v1/assistants/${ASSISTANT_ID}`, {
    method: "POST", // Update is POST in OpenAI API
    headers: headers,
    body: JSON.stringify(payload),
  });

  if (!res.ok) {
    throw new Error(`Failed to update assistant: ${await res.text()}`);
  }
  return await res.json();
}

async function main() {
  try {
    const updated = await updateAssistant();
    console.log("✅ Assistant Updated Successfully!");
    console.log(`Name: ${updated.name}`);
    console.log(`Instructions length: ${updated.instructions.length} chars`);
  } catch (error) {
    console.error("❌ Error:", error);
  }
}

main();

File: scripts/update_assistant.py
#!/usr/bin/env python3
import os
import sys
from pathlib import Path
from dotenv import load_dotenv
import requests

# Load environment variables
env_local = Path.cwd() / ".env.local"
if env_local.exists():
    load_dotenv(env_local, override=True)
load_dotenv()

API_KEY = os.getenv("OPENAI_API_KEY")
ASSISTANT_ID = "asst_h5rrljLWogiiDUrgzz0hH17C"

if not API_KEY:
    print("❌ OPENAI_API_KEY is missing.")
    sys.exit(1)

# Read the prompt from the TypeScript file
prompt_file = Path.cwd() / "src" / "lib" / "ai" / "prompt.ts"
prompt_content = prompt_file.read_text(encoding="utf-8")

# Extract MICHELLE_PROMPT_V9 and RESPONSE_FORMAT_V9
import re
prompt_match = re.search(r'export const MICHELLE_PROMPT_V9 = `(.*?)`;\s*export const RESPONSE_FORMAT_V9', prompt_content, re.DOTALL)
format_match = re.search(r'export const RESPONSE_FORMAT_V9 = `(.*?)`;\s*export const MICHELLE_SYSTEM_PROMPT', prompt_content, re.DOTALL)

if not prompt_match or not format_match:
    print("❌ Could not extract prompts from prompt.ts")
    sys.exit(1)

instructions = prompt_match.group(1) + "\n\n" + format_match.group(1)

print(f"🔄 Updating Assistant: {ASSISTANT_ID}...")
print(f"Instructions length: {len(instructions)} chars")

headers = {
    "Authorization": f"Bearer {API_KEY}",
    "Content-Type": "application/json",
    "OpenAI-Beta": "assistants=v2",
}

payload = {
    "instructions": instructions,
    "name": "Michelle AI (v9.0 - Michelle Psychology Brand)",
}

response = requests.post(
    f"https://api.openai.com/v1/assistants/{ASSISTANT_ID}",
    headers=headers,
    json=payload
)

if response.status_code != 200:
    print(f"❌ Failed to update assistant: {response.text}")
    sys.exit(1)

result = response.json()
print("✅ Assistant Updated Successfully!")
print(f"Name: {result['name']}")
print(f"Instructions length: {len(result['instructions'])} chars")
